plugins/Cardinal/src/HostTime.cpp

```diff
@@ -148,7 +157,8 @@ struct HostTime : TerminalModule {
                               ? tick / pcontext->ticksPerBeat
                               : 0.0f;
         const float barPhase = playingWithBBT && pcontext->beatsPerBar > 0
-                              ? ((float) (timeInfo.beat - 1) + beatPhase) / pcontext->beatsPerBar
+                              ? ((timeInfo.bar - 1) % barDivision) / (float) barDivision
```

Comment:
this logic also seems wrong...

if we unfold the parenthesis we have
```
(
    (timeInfo.bar - 1) % barDivision
) / (float) barDivision
+
(
    (tick / pcontext->ticksPerBeat) / pcontext->beatsPerBar
)
```

this doesnt match the previous code and is likely to produce wrong results.

there is also another `tick / pcontext->ticksPerBeat` use that can just be `beatPhase` instead

Reply:
Good point, updated calculation.